cmd/skopeo/sync.go

```diff
@@ -278,21 +279,82 @@ func imagesToCopyFromRegistry(registryName string, cfg registrySyncConfig, sourc
 		serverCtx.DockerAuthConfig = &cfg.Credentials
 
 		var sourceReferences []types.ImageReference
-		for _, tag := range tags {
-			source := fmt.Sprintf("%s:%s", repoName, tag)
 
-			imageRef, err := docker.ParseReference(source)
+		switch tags.(type) {
+		case []string, []interface{}, nil:
+			tagList := make([]string, 0)
+			if tagIns, ok := tags.([]interface{}); ok {
+				for _, tagValue := range tagIns {
+					switch tagValue.(type) {
+					case string, int, float64:
```

Comment:
The `int`/`float64` types in here are scary. Are we really doing tag matches on a roundtrip between string/binary float/string?